File: portfolio-backend/src/main/java/com/raama/portfoliobackend/entity/Experience.java
package com.raama.portfoliobackend.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.sql.Date;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import lombok.Getter;
import lombok.Setter;

@Getter @Setter
@Entity
public class Experience {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String position;
    private String company;
    private String logoLoc;
    private String location;
    private Date dateStart;
    private Date dateEnd;
    private String description;
    
    @JsonIgnoreProperties("experiences")
    @ManyToOne(optional = false)
    private Person person;
    
    public Experience(){}

}

File: src/app/models/experience.ts
import { Person } from "./person";

export class Experience {
    id?: number;
    position: string;
    company: string;
    imgUrl: string;
    location: string;
    dateStart: Date;
    dateEnd: Date;
    description: string;
    person: Person;

    constructor(position: string, company: string, imgUrl: string, location: string,
        dateStart: Date, dateEnd: Date, description: string, person: Person){
            this.position = position;
            this.company = company;
            this.imgUrl = imgUrl;
            this.location = location;
            this.dateStart = dateStart;
            this.dateEnd = dateEnd;
            this.description = description;
            this.person = person;
    }
}